src/Generator/Program.fs

```diff
@@ -453,26 +483,87 @@ let formatParam (p: TypedParameterInfo) =
     let name = formatName p.name
     sprintf "%s %s" (p.typ |> typeToString) name
 
-let generateInterface enums (functions: TypedFunctionDeclaration[]) =
+let outputPath = "../binding/Binding"
+
+let generateDummyTypes () =
+    use backing = new StringWriter()
+    use writer = new System.CodeDom.Compiler.IndentedTextWriter(backing)
+    let writeLine (str: string) = writer.WriteLine str
+    let indent() = writer.Indent <- writer.Indent + 1
+    let unindent() = writer.Indent <- writer.Indent - 1
+    let usings = 
+        [
+            "System"
+        ]
+    for using in usings do
+        sprintf "using %s;" using
+        |> writeLine
+
+    writeLine ""
+    writeLine "namespace FooBar"
+    writeLine "{"
+    indent()
+    
+    let formatDummyType ty =
+        sprintf "public struct %s {}" ty.name
+
+    for ty in additionalTypesToGenerate do
+        match ty._namespace with
+        | Some n ->
+            sprintf "namespace %s" n
+            |> writeLine
+            writeLine "{"
+            indent()
+            formatDummyType ty
+            |> writeLine
+            unindent()
+            writeLine "}"
+        | None ->
+            formatDummyType ty
+            |> writeLine
+
+    unindent()
+    writeLine "}"
+    backing.Flush()
+    File.WriteAllText(outputPath + "/" + "Types.cs", backing.ToString())
+    
+
+let generateEnums enums =
     use backing = new StringWriter()
```

Comment:
(I am not sure whether this review editor grabs the lines i selected... if not.... line 532 to 536... consider pulling these writers, writeline, indent, unindent out, since this block of code is copied multiple times.. Maybe some `Formatter` providing functions for indent/unindent/writeline and writer is internal to `Formatter`

Reply:
The formatter was minimal refactored to produce statements which are then aggregated in a whole batch.